Modules/SubModule_ASL/xASL_wrp_ResampleASL.m

```diff
@@ -222,18 +223,38 @@ function xASL_wrp_ResampleASL(x)
     elseif round(dim4/2)~=dim4/2
         warning('Odd number of control-label pairs, skipping');
         return;
-    else    
+    else
+        
         % Paired subtraction
         [ControlIm, LabelIm] = xASL_quant_GetControlLabelOrder(ASL_im);
         ASL_im = ControlIm - LabelIm;
-        % Save PWI4D
-        fprintf('%s', [PathPWI4D{iSpace} ', ']);
-        xASL_io_SaveNifti(PathASL{iSpace}, PathPWI4D{iSpace}, ASL_im, 32, false);
+        % preallocate PWI
+        PWI = zeros(size(ASL_im,1),size(ASL_im,2),size(ASL_im,3));

```

Comment:
What happens here if `size(ASL_im,4)~=1`?

Reply:
I've tested it, it works fine. The result would be that we average the ASL_im over one repeat and that works, the PWI that is create is a single 3D volume